### DISCUSSION TITLE
Extensions based on nice gui

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#1817] [Ideas / Feature Requests]
if we could introduce extensions which could be installed when needed ,like the mkdocs extension.there could be some official extensions and also plenty of extensions by thirty.   There are extension about websites template ,or some specific extensions

**Comment by rodja** [comment]
Sure. Extensions are simply python packages which depend on NiceGUI and add some new stuff. Like [CatDesign](https://github.com/ofenbach/CatDesign). Or have you something other in mind?

**Reply by v479038280** [reply]
not only some widgets based on nicegui, and also themes ,we may get some ideas from mkdocs

**Reply by v479038280** [reply]
https://github.com/streamsync-cloud/streamsync
with such tool we don't need to check the classes everytime, I always need to refresh the knowledge of attriute in classes when needed.
here is a picture basedon streamsync ,the default appeance is very beautiful ,Maybe a extension with such default layout is very usefull
![微信截图_20231018233146](https://github.com/zauberzeug/nicegui/assets/25096045/6ddf3346-93cd-45eb-9b5f-2a67af2aa3b1)

**Comment by falkoschindler** [comment]
See https://github.com/zauberzeug/nicegui/discussions/1618 for a similar discussion.
